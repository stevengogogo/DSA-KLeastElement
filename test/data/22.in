10 50000
-6314 3620 -7165 5413 -6603 -5689 9133 2749 -7840 -3338
Delete 8
Reverse 3 6
Reverse 3 9
Delete 7
Reverse 5 7
Delete 2
Insert 1 6432
Insert 3 1987
Insert 1 2288
Query 8 8 1
Delete 10
Delete 3
Insert 6 8879
Insert 3 2463
Insert 8 6827
Insert 5 -6490
Insert 11 2859
Insert 7 5110
Delete 11
Reverse 1 6
Reverse 8 9
Insert 9 8900
Insert 14 -8232
Query 6 14 9
Query 12 13 1
Reverse 8 10
Delete 15
Query 11 12 1
Delete 10
Reverse 10 11
Reverse 4 7
Query 11 13 3
Reverse 7 13
Delete 4
Insert 13 -4705
Insert 13 -2874
Delete 1
Query 7 12 5
Reverse 2 3
Query 9 9 1
Reverse 10 13
Delete 4
Query 10 12 2
Insert 7 -8449
Reverse 8 11
Query 7 8 2
Delete 1
Delete 2
Delete 5
Query 4 5 1
Delete 2
Reverse 4 8
Delete 4
Reverse 6 8
Query 3 4 2
Query 1 5 3
Delete 2
Insert 7 4784
Delete 3
Reverse 6 6
Delete 2
Reverse 3 4
Insert 5 1939
Query 1 7 5
Query 1 3 3
Insert 7 1578
Query 1 4 2
Insert 3 -6862
Insert 3 2506
Delete 5
Insert 9 7265
Delete 10
Insert 7 1161
Query 1 7 5
Insert 5 8735
Query 10 11 2
Insert 5 -5028
Reverse 8 12
Delete 9
Insert 6 5856
Insert 3 -7609
Query 3 3 1
Insert 3 -111
Query 3 10 2
Insert 6 -377
Delete 10
Delete 8
Query 11 12 2
Delete 2
Delete 6
Query 9 11 1
Reverse 9 10
Query 7 7 1
Delete 3
Query 3 6 2
Reverse 9 9
Delete 7
Insert 7 -9162
Query 6 7 2
Insert 7 -5781
Delete 10
Reverse 1 3
Reverse 1 10
Insert 5 -5533
Insert 11 8421
Reverse 1 11
Query 5 6 1
Query 9 10 1
Insert 3 -4911
Reverse 13 13
Reverse 5 7
Reverse 2 5
Insert 14 8923
Delete 4
Query 8 9 2
Reverse 12 12
Reverse 13 13
Query 10 11 2
Delete 2
Reverse 4 10
Delete 8
Reverse 11 11
Query 3 8 3
Reverse 8 11
Query 11 11 1
Delete 8
Reverse 4 5
Query 2 2 1
Query 9 9 1
Delete 2
Delete 3
Reverse 5 5
Delete 2
Delete 6
Insert 5 -7191
Insert 7 -6570
Reverse 8 8
Delete 6
Reverse 5 7
Reverse 7 7
Insert 1 2600
Delete 3
Delete 1
Reverse 6 6
Delete 5
Delete 5
Query 1 4 4
Query 2 4 1
Insert 3 2248
Delete 1
Query 4 4 1
Insert 3 9246
Delete 3
Insert 5 9986
Insert 2 6212
Delete 5
Query 2 2 1
Reverse 5 5
Delete 2
Insert 3 1200
Query 5 5 1
Query 4 5 1
Reverse 5 5
Query 4 4 1
Delete 4
Query 1 2 2
Query 2 3 2
Query 3 4 2
Reverse 1 4
Delete 4
Insert 1 -4248
Insert 4 -8163
Reverse 5 5
Query 4 4 1
Reverse 1 5
Query 3 5 1
Delete 4
Query 3 3 1
Delete 1
Delete 1
Reverse 1 2
Reverse 1 2
Reverse 2 2
Insert 2 1886
Query 1 1 1
Delete 3
Insert 1 389
Reverse 2 2
Insert 2 -8334
Delete 4
Query 1 3 2
Reverse 2 2
Reverse 1 3
Delete 2
Query 2 2 1
Reverse 2 2
Reverse 1 1
Reverse 2 2
Reverse 2 2
Insert 3 -9745
Query 3 3 1
Reverse 1 1
Delete 3
Delete 1
Query 1 1 1
Delete 1
