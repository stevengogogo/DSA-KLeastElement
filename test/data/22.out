-7165
8900
-7165
2859
6827
6827
8900
-2874
-2874
-2874
6827
2859
2288
8900
-2874
2288
7265
-7609
-7609
4784
1161
-7840
-377
-7840
-7840
-9162
-5781
1939
-377
-377
2288
5856
8421
-9162
-377
6212
9986
-5781
-5781
2248
2248
9986
-8163
-4248
1200
1200
389
389
-9745
389
